docs/workflows.md

```diff
@@ -0,0 +1,63 @@
+
+#### Trust registry
+
+As a client you can retrieve the trust registry (and NOT alter it). That is intential and all a consumer/client should and is able to do. Altering the trust registry is baked into admin actions and only possible with admin role.
+
+#### Webhooks
+
+A client can subscribe to the webhooks via the cloudapi (as opposed to directly via the webhooks container). This requires only targetting the `/webhooks` endpoint and optionally targetting the `/webhooks/{topic}` subroute by suffixing a topic. Using the auth mechanism (see section below), the app automatically extracts the required info about the wallet (i.e the wallet id and JWT) and retrieves only the associated webhooks with a particular wallet. Failing to authenticate will return a 403 HTTP Error.
+
+### Authentication
+
+Authentication is handled by the CloudAPI and, fom a client perspective, kept simple and convenient. Either, via the Swagger UI auth (padlock butten in UI) or via the HEADER data of your client specifying an `x-api-key`. Regardless of whether ou use the UI or another client, the `x-api-key` value consists of two parts, separated by a dot:
+
+`{role}.{key/password}`
+
+So, your header has the format `'x-api-key: {role}.{key/password}` which yields, for example, `'x-api-key: yoma.adminApiKey'`
+
+The first part `role` specifies the role on the surface and targets the correct endpoints under the hood and authentication mechanisms under the hood. The CloudAPI knows how to interpret the roles and will produce the correct target URLs for eg aca-py (`member` targets the multitenant agent) with the correct header expected by aca-py agent. For instance, `member` results in a `Bearer {TOKEN}` header against the multitenant agent whereas `member-admin` as role results in an `x-api-key` header for the multitenant agent (hence targeting the admin interface of the same multitennat agent). You may have noticed now that this mechanism also chooses which aca-py instance to target without having to know or specify the URL the agent resides under.
+
+Currently there are five options for `role`:
+
+- yoma
+- ecosystem
+- ecosystem-admin
+- member
+- member-admin
+
+the `yoma` and `-admin` suffixed roles are admin roles. The rest are non-admin roles meaning non-admin roles have no exposure to the aca-py admin tasks nor any documented endpoints prefixed `admin:` in the cloudapi.
+
+For admin roles pass the admin password as the second part of `{role}.{key/password}`. For member/ecosystem (non-admin roles) pass the wallets JWT as the second part of `{role}.{key/password}`.
+
+### Creating schemas
```

Comment:
I like that you're going into detail on how these tasks work. What I'm missing is a bit of guidance on how to build an application on this stack. What do you need to build yourself, how do you manage listening to webhooks? What are the steps involved to issue a credential and what do you need to do be able to issue a credential?